oracle/datadog_checks/oracle/oracle.py

```diff
@@ -165,44 +142,39 @@ def check(self, _):
-        """Creates a connection or raises an exception"""
         if self._cached_connection is None:
-            if self.can_use_oracle_client():
-                self._cached_connection = self._oracle_client_connect()
-            elif JDBC_IMPORT_ERROR:
-                self._connection_errors += 1
+            if self.can_use_jdbc():
+                try:
+                    self._cached_connection = self._jdbc_connect()
+                except Exception as e:
+                    self.log.error("The JDBC connection failed with the following error: %s", str(e))
+                    self._connection_errors += 1
+            else:
+                self._cached_connection = self._oracle_connect()
+        return self._cached_connection
+
+    def can_use_jdbc(self):
+        if self._jdbc_driver:
+            if JDBC_IMPORT_ERROR is not None:
                 self.log.error(
-                    "Oracle client is unavailable and the integration is unable to import JDBC libraries. You may not "
+                    "The integration is unable to import JDBC libraries. You might not "
                     "have the Microsoft Visual C++ Runtime 2015 installed on your system. Please double check your "
                     "installation and refer to the Datadog documentation for more information."
                 )
                 raise JDBC_IMPORT_ERROR
             else:
-                self._cached_connection = self._jdbc_connect()
-        return self._cached_connection
-
-    def can_use_oracle_client(self):
-        try:
-            # Check if the instantclient is available
-            cx_Oracle.clientversion()
-        except cx_Oracle.DatabaseError as e:
-            # Fallback to JDBC
-            self.log.debug('Oracle instant client unavailable, falling back to JDBC: %s', e)
-            return False
-        else:
-            self.log.debug('Running cx_Oracle version %s', cx_Oracle.version)
-            return True
+                return True
 
-    def _oracle_client_connect(self):
+    def _oracle_connect(self):
         dsn = self._get_dsn()
-        self.log.debug("Connecting via Oracle Instant Client with DSN: %s", dsn)
+        self.log.debug("Connecting to Oracle with DSN: %s", dsn)
         try:
-            connection = cx_Oracle.connect(user=self._user, password=self._password, dsn=dsn)
-            self.log.debug("Connected to Oracle DB using Oracle Instant Client")
+            connection = oracledb.connect(user=self._user, password=self._password, dsn=dsn)
+            self.log.debug("Connected to Oracle DB using Python Oracle")
             return connection
-        except cx_Oracle.DatabaseError as e:
+        except oracledb.DatabaseError as e:
             self._connection_errors += 1
-            self.log.error("Failed to connect to Oracle DB using Oracle Instant Client, error: %s", str(e))
+            self.log.error("Failed to connect to Oracle DB using Python Oracle, error: %s", str(e))
```

Comment:
```suggestion
            self.log.error("Failed to connect to Oracle DB, error: %s", str(e))
```
Since this is not as explicit as the instant client was I think we could hide the implementation detail